worker/rhpv3.go

```diff
@@ -458,10 +458,16 @@ func (a *account) WithWithdrawal(ctx context.Context, amtFn func() (types.Curren
 		}
 		return err
 	}
-	if err != nil {
+	// if the amount is zero we just retur the error
+	if amt.IsZero() {
```

Comment:
it would probably be more logical to do:

```go
if err != nil && isBalanceInsufficient(err) {
    ...
    return err
} else if err != nil {
    return err
}

if amt.IsZero() {
    return nil
}
return a.bus.AddBalance(ctx, a.id, a.host, new(big.Int).Neg(amt.Big()))
```

or make it a big `if else if else` block

Reply:
Updated it. Let me know if it's more readable now.